pkg/api/controller.go

```diff
@@ -3223,6 +3223,8 @@ func (c *Controller) UpdatePassword(w http.ResponseWriter, r *http.Request, body
 		writeError(w, http.StatusInternalServerError, http.StatusText(http.StatusInternalServerError))
 		return
 	}
+	// RemoveUserFromCache to clean the cache from the old password
+	c.Auth.RemoveUserFromCache(user.Username)
```

Comment:
I think that if we call it part of update password
- it will prevent future issues as it came to solve this specific problem
- the user that call the auth API doesn't aware of any cache, and now we expose implementation detail as part of the interface